src/Core/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -360,10 +361,12 @@ protected static object ParseParamAsSystemType(string param, Type systemType)
                 "Double" => double.Parse(param),
                 "Decimal" => decimal.Parse(param),
                 "Boolean" => bool.Parse(param),
-                "DateTime" => DateTimeOffset.Parse(param),
+                "DateTime" => DateTime.Parse(param, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeUniversal),
                 "DateTimeOffset" => DateTimeOffset.Parse(param),
                 "Date" => DateOnly.Parse(param),
                 "Guid" => Guid.Parse(param),
+                "TimeOnly" => TimeOnly.Parse(param),
+                "TimeSpan" => TimeOnly.Parse(param),
```

Comment:
This needs a test to confirm behavior: TimeSpan can be a negative number https://learn.microsoft.com/en-us/dotnet/api/system.timespan?view=net-7.0#:~:text=A%20TimeSpan%20value%20can,a%20time%20of%20day.

so if a param value is set to `-10.20:30:40.0500000` won't this fail? There is some historical precedent with entity framework and sql time type the relies on the user to provide a valid value within 0-24 hours, but I would rather not introduce that ambiguity here. 

Check out this doc to see if there are any pointers: https://learn.microsoft.com/en-us/dotnet/standard/datetime/how-to-use-dateonly-timeonly#work-with-timespan-and-datetime

Reply:
SqlClient currently is not recognizing type TimeOnly, the only way to use TimeOnly is to Parse it. That's why I have added Both, But the parsing will only happen into TimeOnly.

I have added a test for negative values as well.

Reply:
> SqlClient currently is not recognizing type TimeOnly,

This works with SqlClient 5.1.1 though right?

Reply:
Sql 5.1.1 supports TimeOnly, but the problem we have is the `SqlDataReader` which we use to parse the Schema of tables. in that the SystemType automatically comes as TimeSpan, instead of TimeOnly for SqlDbType.Time